### ISSUE TITLE
nonlinsolve has trouble getting full solution

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#22723]
cf #22717 
From this [SO post](https://stackoverflow.com/questions/70413564/sympy-seems-to-fail-solving-a-system-of-non-linear-equations-while-being-able-to/70416346#70416346):
```python
>>> eqs = Tuple(-y**2 + log(y**2/x) + 2, -2*x*y + 2*x/y)
>>> nonlinsolve((-y**2 + log(y**2/x) + 2, -2*x*y + 2*x/y), (x, y))
FiniteSet((Complement({y**2*exp(2 - y**2)}, {0}), y))
```
If `x` and `y` are real, then `y**2*exp(2 - y**2)` is obtained for `x`. This solution sets the first equation to 0 but not the second except for values of `y` equal to +/-1. The manual option for solve resolves the solutions:
```python
>>> solve((-y**2 + log(y**2/x) + 2, -2*x*y + 2*x/y), manual=True)
[{y: -1, x: E}, {y: 1, x: E}]
```

**Comment by soma2000-lang** [comment]
@smichr  Can I work on this isssue?Please assign me this.

**Comment by smichr** [comment]
> Please assign me this.

Yes. There is no need to ask. You can work on anything you wish. If someone says that they want to work on an issue, however, you might want to give them a few days before creating your own PR.

**Comment by smichr** [comment]
@soma2000-lang , I see that I have a PR #22719 for the solve-related issues. If you could review that it can be committed and then if you want to look at how to implement a similar change in `solveset` that would be a good place to start.
